src/AutoRest.CSharp/Generation/Writers/RestClientWriter.cs

```diff
@@ -243,6 +244,18 @@ private void WriteRequestCreation(CodeWriter writer, RestClientMethod clientMeth
                             flattenedSchemaRequestBody.Serialization,
                             w => w.Append(modelVariable));
                         break;
+                    case UrlEncodedBody urlEncodedRequestBody:
+                        WriteHeaders(writer, clientMethod, request, content: true);
+                        writer.Append($"var contentBuilder = new Azure.Core.FormUrlEncodedContent.Builder ();\n");
+
+                        foreach (var (name, value) in urlEncodedRequestBody.Values)
+                        {
+                            writer.Append($"contentBuilder.Add(\"{name}\",");
+                            WriteConstantOrParameter(writer, value);
+                            writer.Line($"{(value.Type.IsNullable ? "?" : "")}.ToString());");
+                        }
```

Comment:
Use the `asString` parameter of the `WriteConstantOrParameter`  here.

Reply:
WriteConstantOrParameter has a `bool enumAsString = false but my type is not necessarily an enum?

Reply:
Good point. Let's add WriteConstantOrParameterAsString method and encapsulate logic there `ToString`. But we should still use `enumAsString`.

Reply:
https://github.com/Azure/autorest.csharp/pull/1042/files#r578808631 thats why.